extensions/interactions/MultipleChoiceInput/protractor.js

```diff
@@ -36,13 +36,21 @@ var customizeInteraction = function(elem, richTextInstructionsArray) {
 // the options.
 var expectInteractionDetailsToMatch = function(
     elem, richTextInstructionsArray) {
-  elem.all(by.repeater('choice in $ctrl.choices track by $index'))
+  elem.all(by.repeater('choice in $ctrl.choices'))
     .then(function(optionElements) {
+      var choices = [];
       expect(optionElements.length).toEqual(richTextInstructionsArray.length);
       for (var i = 0; i < optionElements.length; i++) {
-        forms.expectRichText(optionElements[i].element(by.css(
+        optionElements[i].element(by.css(
           '.protractor-test-multiple-choice-option'
-        ))).toMatch(richTextInstructionsArray[i]);
+        )).getText().then(function(text) {
+          choices.push(text);
+          if (choices.length === richTextInstructionsArray.lengt) {
```

Comment:
"length" is misspelled -- but more seriously, this means that your e2e test does not work. In addition to fixing this, please reply to this comment with an analysis of why this part of the code was not triggered and an explanation of what you have done to fix it. 

I also don't see why you need an "if" statement here -- this should be replaced by an expectation that the lengths match. In general, please write the tests without any "if" statements. Generally, there shouldn't be conditional logic in tests.

Reply:
When I corrected the mistake, all seems to work fine. After the correction I verified that the code inside the if block was executed. The compiler simply did not flag the error previously because choice.lengt resulted in an undefined which is still a valid ans.
Here and if statement so that the following code is executed only when there are 4 items in the choice list(created by me). Since it is created for temporary use there is no sense in using expect which is going to flag an error three times.

Reply:
The explanation of the issue with lengt is satisfactory -- thanks for that.

However, you still cannot do the "if". This is a utility file. It is a file that is meant for general-purpose use, and should therefore not have an artificial restriction to the case of 4 items. It should work and do the same sorts of checks, regardless of how many items there are in the list. Does that make sense?

Reply:
I have checked it for 5 inputs by changing the list length and it still works. The only reason for the if block is that the choice list is gotten asynchronously and it can be executed only after all the inputs are fetched. The length comparison is already been checked in a previous check.

Reply:
Ah, I see. OK, I get what you're doing now -- thanks for explaining.

However, it is still a little indirect. It would be better to explicitly do the final check after all promises have been resolved. Do any of the methods on [this SO page](https://stackoverflow.com/questions/24852126/is-there-a-way-to-resolve-multiple-promises-with-protractor) or similar work?

Reply:
I have made the necessary changes. Done.